src/v/pandaproxy/rest/rest_request_auth.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/rest/rest_request_auth.h"
+
+#include "config/configuration.h"
+#include "pandaproxy/logger.h"
+#include "seastar/http/exception.hh"
+#include "security/scram_algorithm.h"
+
+namespace pandaproxy {
+
+rest_request_authenticator::rest_request_authenticator(
+  config::binding<bool> require_auth, cluster::controller* controller)
+  : _controller(controller)
+  , _require_auth(std::move(require_auth)) {}
```

Comment:
Remote require auth. It's only true when http_basic is enabled and that check is done in the handlers.